Laravel 5.4

jpuck opened the pull request:
If you merge this into master, then you should bump the major version (2.0) to comply with [semver][1] because there were quite a few breaking changes that came about with 5.3 and 5.4.

Just to get started, service providers are initialized with a different method now 3366b33

With the introduction of the built-in web and api middleware routes, the session variable is only attached to the web routes (conversely api routes use tokens), so that needed to be loaded differently in 275b9ba in order for a user to stay logged in for subsequent requests.

Fixed the jwt-auth dependencies #24, including bootstrapping the aliases inside the service provider in 	8a806bb

I wrote this to support multiple authentication types aside from just shib, and I noticed there was a bug when falling back to native authentication due to the `type` value not being set in 50fdb57

There have been some general namespacing conventions lately inside the config files, so I conformed the `group_model` and user `model` and revamped the documentation with Github-style markdown for syntax highlights.

[1]:http://semver.org/

jpuck committed:
doc class syntax with highlights

jpuck committed:
doc composer

jpuck committed:
fix laravel 5.4 extend provider

jpuck committed:
update database migrations

jpuck committed:
require jwt-auth fixes StudentAffairsUWM/Laravel-Shibboleth-Service-Provider#24

use alias loader

jpuck committed:
fix detached session by applying web middleware to routes

jpuck committed:
fix mixed type authentication var not set

jpuck committed:
tidy consistency of config file options

jpuck committed:
bump version

jpuck commented:
fixed in https://github.com/StudentAffairsUWM/Laravel-Shibboleth-Service-Provider/pull/25/commits/3366b3342126abc6aeb65c56bcf112cf5586cd5b

cjmaio commented:
Thank you for this commit! It's really appreciated. I am going to try and test this out within the next day or so, but assuming all goes well I'll be merging this in.

jpuck committed:
fix PSR-2 coding standards automatically with PHP_CodeSniffer

jpuck committed:
fix cs line length warnings

jpuck committed:
add code climate, ignore facades & migration up()

jpuck committed:
remove unused code

jpuck committed:
remove cs else statements

jpuck committed:
remove cs unused variables

jpuck committed:
Merge coding standards fixes

jpuck committed:
fix consistent variable naming convention

jpuck committed:
move redirect outside conditional

fixes StudentAffairsUWM/Laravel-Shibboleth-Service-Provider#26

jpuck commented:
* fixed a lot of [PSR-2][2] coding standards and smells with [phpmd][1] and [PHP_CodeSniffer][3] along with a badge for [code climate][4].
* added support for lazy shib closing #26 

[1]:https://phpmd.org/
[2]:http://www.php-fig.org/psr/psr-2/
[3]:https://github.com/squizlabs/PHP_CodeSniffer
[4]:https://lima.codeclimate.com/github/uawcob/Laravel-Shibboleth-Service-Provider/compare/lara5.4

jpuck committed:
fix typo case-sensitive view

jpuck committed:
doc specific vendor publish

jpuck committed:
fix obsolete hard-coded version

jpuck committed:
use exception class

jpuck committed:
test lint php on travis

jpuck committed:
fix backwards compat php 5.4 empty expression

fixes StudentAffairsUWM/Laravel-Shibboleth-Service-Provider#28

jpuck committed:
fix backwards compat php 5.4 ::class
